client/api/src/main/java/org/opennms/velocloud/client/api/model/Edge.java

```diff
@@ -38,12 +40,45 @@ public class Edge {
     public final String operator;
     public final String site;
     public final boolean hub;
+    public final List<Link> links;
+    public final Boolean alertsEnabled;
+    public final String buildNumber;
+    public final String customInfo;
+    public final String description;
+    public final String deviceFamily;
+    public final String deviceId;
+    public final String dnsName;
+    public final String lteRegion;
+    public final String logicalId;
+    public final String modelNumber;
+    public final String name;
+    public final Boolean operatorAlertsEnabled;
+    public final String selfMacAddress;
+    public final Integer siteId;
+    public final String softwareVersion;
 
     private Edge(final Builder builder) {
         this.enterpriseId = Objects.requireNonNull(builder.enterpriseId);
         this.operator = Objects.requireNonNull(builder.operator);
         this.site = Objects.requireNonNull(builder.site);
         this.hub = builder.hub;
+        this.links = Objects.requireNonNull(builder.links);
+        this.alertsEnabled = Objects.requireNonNull(builder.alertsEnabled);
+        this.buildNumber = Objects.requireNonNull(builder.buildNumber);
+        this.customInfo = Objects.requireNonNull(builder.customInfo);
+        this.description = Objects.requireNonNull(builder.description);
+        this.deviceFamily = Objects.requireNonNull(builder.deviceFamily);
+        this.deviceId = Objects.requireNonNull(builder.deviceId);
+        this.dnsName = Objects.requireNonNull(builder.dnsName);
+        this.lteRegion = Objects.requireNonNull(builder.lteRegion);
+        this.logicalId = Objects.requireNonNull(builder.logicalId);
+        this.modelNumber = Objects.requireNonNull(builder.modelNumber);
+        this.name = Objects.requireNonNull(builder.name);
+        this.operatorAlertsEnabled = Objects.requireNonNull(builder.operatorAlertsEnabled);
+        this.selfMacAddress = Objects.requireNonNull(builder.selfMacAddress);
+        this.siteId = Objects.requireNonNull(builder.siteId);
+        this.softwareVersion = Objects.requireNonNull(builder.softwareVersion);
```

Comment:
Nothing in the `EdgeObject` is marked as required, so I'm not sure which of these we can count on to be populated.

Reply:
I think we just need to play by ear and once we are able to test against the dev environment we'll figure that out. I did everything required but from the API documentation doesn't seem that many fields are.

Reply:
I think having this as required is fine for now.